### DISCUSSION TITLE
Autocomplete - How to remove symbols from completion?

### DISCUSSION THREAD

**Discussion by P-p-H-d** [microsoft/vscode-cpptools#7058] [Q&A] [answered]
My use case is a C header only library.

Some symbols of theses libraries (macros, types, ...) are only for internal use and I would like then not to appear when the user of the library writes its own code.

How can I "remove" symbols from the autocomplete database (or at least order then)?
Is there a unique predefined macro that only exists when the parser analyzes the code?
Thanks

**Comment by michelleangela** [comment, marked as answer]
Currently there isn't a setting to remove certain symbols. There's a feature request to add a completion list filter option (https://github.com/microsoft/vscode-cpptools/issues/1798). @sean-mcmanus are there workarounds for this?

**Reply by sean-mcmanus** [reply]
Yeah, you can surround the stuff you want to exclude with

```
#ifndef __INTELLISENSE__
#endif
```
![image](https://user-images.githubusercontent.com/19859882/109359021-98fb3100-7839-11eb-903e-5af1b89ead8a.png)
